graphql_compiler/cost_estimation/cardinality_estimator.py

```diff
@@ -69,17 +69,99 @@ def _get_last_edge_direction_and_name_to_location(location):
     return edge_direction, edge_name
 
 
-def _get_parent_and_child_name_from_edge(schema_graph, child_location):
+def _get_parent_and_child_base_classnames_from_edge(schema_graph, child_location):
     """Get the base classname of a location and its parent from the last edge information."""
     edge_direction, edge_name = _get_last_edge_direction_and_name_to_location(child_location)
     edge_element = schema_graph.get_edge_schema_element_or_raise(edge_name)
     if edge_direction == INBOUND_EDGE_DIRECTION:
-        parent_name_from_edge = edge_element.base_out_connection
-        child_name_from_edge = edge_element.base_in_connection
+        parent_base_classname = edge_element.base_out_connection
+        child_base_classname = edge_element.base_in_connection
     elif edge_direction == OUTBOUND_EDGE_DIRECTION:
-        parent_name_from_edge = edge_element.base_in_connection
-        child_name_from_edge = edge_element.base_out_connection
-    return parent_name_from_edge, child_name_from_edge
+        parent_base_classname = edge_element.base_in_connection
+        child_base_classname = edge_element.base_out_connection
+    else:
+        raise AssertionError(u'Expected edge direction to be either inbound or outbound.'
```

Comment:
Why not "Expected edge direction to be either 'in' or 'out'." ? It communicates the same message and is just as precise.

Reply:
Edge direction is something the user specifies, but only implicitly. For example, we don't have a field called `edge direction: ` which the user fills with either `in` or `out`. If the error message were `either 'in' or 'out'`, the user would get the impression that they've missed an `edge direction: ` field somewhere.